bodhi/server/services/updates.py

```diff
@@ -583,3 +592,41 @@ def waive_test_results(request):
         request.errors.add('body', 'request', str(e))
 
     return dict(update=update)
+
+
+@update_get_test_results.get(schema=bodhi.server.schemas.GetTestResultsSchema,
+                             validators=(validate_update_id,
+                                         validate_acls),
+                             permission='read', renderer='json',
+                             error_handler=bodhi.server.services.errors.json_handler)
+def get_test_results(request):
+    """
+    Get the test results on a given update when gating is on.
+
+    Args:
+        request (pyramid.request): The current request.
+    Returns:
+        dict: A dictionary mapping the key "update" to the update.
+    """
+    update = request.validated['update']
+
+    decision = None
+    try:
+        decision = update.get_test_gating_info()
+    except RequestsTimeout as e:
+        log.exception("Error querying greenwave for test results - timed out")
+        request.errors.add('body', 'request', str(e))
+        request.errors.status = 504
+    except RequestException as e:
+        log.exception("Error querying greenwave for test results")
+        request.errors.add('body', 'request', str(e))
+        request.errors.status = 502
```

Comment:
```call_api()``` will raise a ```RuntimeError``` if Greenwave returns a non-200 code. We should catch that and turn it into a ```502```.

Reply:
done